site/lib/src/components/pages/changelog/changelog_index.dart

```diff
@@ -0,0 +1,109 @@
+import '../../../markdown/markdown_parser.dart';
+
+final class ChangelogIndex extends StatelessComponent {
+  const ChangelogIndex({super.key});
+
+  @override
+  Component build(BuildContext context) {
+    // Reset static filters state to prevent SSR state leaks in dev mode.
+    ChangelogFiltersSidebar.filters.disposeState(notify: false);
+
+    final changesData = context.page.data['changelog'] as List<Object?>?;
+
+    final changelogEntries = <ChangelogEntry>[];
+    if (changesData != null) {
+      for (final change in changesData) {
+        changelogEntries.add(
+          ChangelogEntry.fromMap(change as Map<String, Object?>),
+        );
+      }
+    }
+
+    return div(id: 'changelog-index-content', [
+      div(classes: 'left-col', id: 'changelog-main-content', [
+        const ChangelogFilters(),
+        div(id: 'all-changelog-list', [
+          for (final item in changelogEntries) _ChangelogEntryCard(item),
+        ]),
+      ]),
+      const ChangelogFiltersSidebar(),
+    ]);
+  }
+}
+
+class _ChangelogEntryCard extends StatelessComponent {
+  const _ChangelogEntryCard(this.entry);
+
+  final ChangelogEntry entry;
+
+  @override
+  Component build(BuildContext context) {
+    // Generate a somewhat unique ID for DOM filtering
+    final uniqueId = slugify(
+      '${entry.version}-${entry.area}-${entry.subArea ?? ''}-${entry.description.substring(0, min(20, entry.description.length))}-${entry.hashCode}',
+    );
+
+    return div(
+      classes: 'changelog-card card',
+      id: uniqueId,
+      attributes: {
+        'data-version': entry.version,
+        if (entry.releaseDate != null) 'data-releasedate': entry.releaseDate!,
+        'data-area': entry.area,
+        if (entry.subArea != null) 'data-subarea': entry.subArea!,
+        'data-tags': entry.tags.map((t) => t.id).join(','),
+        // Store description for client-side search.
+        // Warning: This might be large.
+        'data-description': entry.description,
+        if (entry.link != null) 'data-link': entry.link!,
+      },
+      [
+        div(classes: 'card-header', [
+          div(classes: 'header-left', [
+            span(classes: 'version-badge', [text(entry.shortVersion)]),
+
+            span(classes: 'area-badge', [
+              text(
+                entry.area +
+                    (entry.subArea != null ? ': ${entry.subArea}' : ''),
+              ),
+            ]),
+          ]),
+          div(classes: 'tags', [
+            for (final tag in entry.tags)
+              span(classes: 'tag-label ${tag.id}-tag', [text(tag.label)]),
+          ]),
+        ]),
+        div(classes: 'card-content', [
+          if (entry.releaseDate != null)
+            p(classes: 'release-date', [
+              em([text('Released: ${entry.releaseDate} • v${entry.version}')]),
+            ]),
+          DashMarkdown(content: entry.description),
+          if (entry.link != null)
+            div(classes: 'read-more', [
```

Comment:
Consider updating this to use the already existing `Button` component. If not, consider adding the `text-button` class to make the styling and interaction handling more consistent.

Reply:
Updated!